docs/developers/protocol/technical-references/market-order/README.md

```diff
@@ -0,0 +1,364 @@
+
+:::info Example 1
+A taker calls `marketOrderByTick` on the offer list with:
+
+* `maxTick` = -75000
+  * corresponds to a max ratio of 0.0005539
+* `fillVolume` = 2,500 * 10**18
+* `fillWants` = `true`
+
+Since `fillWants = true`, we have:
+
+* `fillVolume` is in `olKey.outbound_tkn` and corresponds to 2,500 DAI
+* the market order will be considered filled once 2,500 DAI has been received.
+
+In summary, this corresponds to a **buy** order for 2,500 DAI and the taker is willing to pay up to `2,500 * 0.0005539 = 1.3832` WETH.
+
+The market order will execute as follows:
+
+1. Get 925.26 DAI for `925.26 * 0.0005476 = 0.5067` WETH from offer #77
+2. Get 916.47 DAI for `916.47 * 0.0005476 = 0.5019` WETH from offer #177
+3. Get the remaining `2500 - 925.26 - 916.47 = 658.27` DAI for `658.27 * 0.0005510 = 0.3627` WETH from offer #42.
+
+In total, the taker gets 2,500 DAI and sends `0.5067 + 0.5019 + 0.3627 = 1.3713` WETH.
+:::
+
+:::info Example 2
+A taker calls `marketOrderByTick` on the offer list with:
+
+* `maxTick` = -75000
+  * corresponds to a max ratio of 0.0005539
+* `fillVolume` = 1.2 * 10**18
+* `fillWants` = `false`
+
+Since `fillWants = false`, we have:
+
+* `fillVolume` is in `olKey.inbound_tkn` and corresponds to 1.2 WETH
+* the market order will be considered filled once 1.2 WETH has been sent.
+
+In summary, this corresponds to a **sell** order of 1.2 WETH and the taker wants to receive at least `1.2 / 0.0005539 = 2168.84` DAI.
+
+1. Sell `925.26 * 0.0005476 = 0.5067` WETH for 925.26 DAI to offer #77
+2. Sell `916.47 * 0.0005476 = 0.5019` WETH for 916.47 DAI to offer #177
+3. Sell the remaining `1.2 - 0.5067 - 0.5019 = 0.1914` WETH for `0.1914 / 0.0005510 = 347.37` DAI to offer #42.
+
+In total, the taker gets `925.26 + 916.47 + 347.37 = 2,189.10` DAI and sends 1.2 WETH.
+:::
+
+### More on market order behaviour
+
+Suppose one wants to buy or sell some token `B` (base), using token `Q` (quote) as payment, on a market with tick spacing `T`.
+
+* **Market buy:** A limit **buy** order for `x` `B` tokens, corresponds to a `marketOrderByTick` on the (`B`,`Q`,`T`) offer list with `fillWants` set to `true`, `fillVolume = x` (the volume one wishes to buy), and `maxTick` = $⌊log_{1.0001}(price\ limit)⌋$.
+* **Market sell:** A limit **sell** order for `x` `B` tokens, corresponds to a `marketOrderByTick` on the (`Q`,`B`,`T`) offer list with
+ `fillWants` set to `false`, `fillVolume = x` (the volume one wishes to sell), and `maxTick` = $⌊-log_{1.0001}(price\ limit)⌋$.
+
+
+:::caution **On order residuals**
+
+Contrary to [GTC orders](https://www.investopedia.com/terms/g/gtc.asp) on regular [order book](https://www.investopedia.com/terms/o/order-book.asp) based exchanges, the residual of your order (i.e., the volume you were not able to buy/sell due to hitting your price limit) will _not_ be put on the market as an offer. Instead, the market order will simply end partially filled.
```

Comment:
Can't we add a reference there to limit orders for the GTC equivalent there ?

Reply:
Good idea. Addressed in c4b7075373158abf24e0fa162825a4133e9690a5